src/WTG.Analyzers/Analyzers/LinqEnumerable/LinqEnumerableAnalyzer.cs

```diff
@@ -52,189 +55,205 @@ public static void CompilationStart(CompilationStartAnalysisContext context)
-				case SyntaxKind.ImplicitArrayCreationExpression:
-					return ((ImplicitArrayCreationExpressionSyntax)e)?.Initializer?.Expressions.Count == 1;
-				case SyntaxKind.ObjectCreationExpression:
-					return ((ObjectCreationExpressionSyntax)e)?.Initializer?.Expressions.Count == 1;
-				case SyntaxKind.ArrayCreationExpression:
-					return ((ArrayCreationExpressionSyntax)e)?.Initializer?.Expressions.Count == 1;
-				default:
-					return false;
-			}
-		}
+		static bool ContainsSingleElement(ExpressionSyntax? e) => LinqEnumerableUtils.GetInitializer(e)?.Expressions.Count == 1;
 
 		public static bool LooksLikeAppend(InvocationExpressionSyntax invocation) => (invocation.ArgumentList.Arguments.Count == 1 ? ContainsSingleElement(invocation.ArgumentList.Arguments[0].Expression) : ContainsSingleElement(invocation.ArgumentList.Arguments[1].Expression));
 
 		public static bool LooksLikePrepend(InvocationExpressionSyntax invocation)
 		{
-			var expression = ((MemberAccessExpressionSyntax)invocation.Expression).Expression;
+			var expression = ((MemberAccessExpressionSyntax)invocation.Expression).Expression.TryGetExpressionFromParenthesizedExpression();
 
-			if (expression.IsKind(SyntaxKind.ParenthesizedExpression))
+			if (expression == null)
 			{
-				expression = ((ParenthesizedExpressionSyntax)expression).GetExpression();
+				expression = ((MemberAccessExpressionSyntax)invocation.Expression).Expression;
 			}
 
 			return (invocation.ArgumentList.Arguments.Count == 1 ? ContainsSingleElement(expression) : ContainsSingleElement(invocation.ArgumentList.Arguments[0].Expression));
 		}
 
-		public static bool IsList(SemanticModel model, ObjectCreationExpressionSyntax e) => e.Type.ToString().StartsWith("List", StringComparison.Ordinal) && model.GetTypeInfo(e).Type?.MetadataName == "List`1";
-		public static bool IsIEnumerable(SemanticModel model, ExpressionSyntax e) => model.GetTypeInfo(e).Type?.MetadataName == "IEnumerable`1";
+		public static bool IsMonadicArityCollection (SemanticModel model, ExpressionSyntax e)
+		{
+			var typeSymbol = model.GetTypeInfo(e).Type;
+
+			if (typeSymbol == null)
+			{
+				return false;
+			}
+
+			if (typeSymbol.MetadataName.Length != 0 && typeSymbol.MetadataName[typeSymbol.MetadataName.Length - 1] != '1')
+			{
+				return false;
+			}
 
-		public static void Analyze (SyntaxNodeAnalysisContext context, FileDetailCache cache, bool hasEnumerable)
+			foreach (var baseType in typeSymbol.AllInterfaces)
+			{
+				if (baseType.IsMatch(WellKnownTypeNames.IEnumerable_T))
+				{
+					return true;
+				}
+			}
+
+			return typeSymbol.IsMatch(WellKnownTypeNames.IEnumerable_T);
+		}
+
+		public static void Analyze(SyntaxNodeAnalysisContext context, FileDetailCache cache, bool hasEnumerableAppendMethod)
 		{
-			if (cache.IsGenerated(context.SemanticModel.SyntaxTree, context.CancellationToken) || !hasEnumerable)
+			if (cache.IsGenerated(context.SemanticModel.SyntaxTree, context.CancellationToken) || !hasEnumerableAppendMethod)
```

Comment:
Early return here if  `!hasEnumerableAppendMethod` looks wrong, because can still analyze concatenation of two literals.

Do we have tests for this case?

Reply:
No, we don't have a test case for this yet - I haven't tried doing this analyzer with any other C# language version.